src/include/sof/lib/mailbox.h

```diff
@@ -17,6 +17,8 @@
 #include <stddef.h>
 #include <stdint.h>
 
+#if !defined(__ZEPHYR__) || !CONFIG_LIBRARY
```

Comment:
Not sure why we need to fake the mailbox stuff, but I guess that SOF needs certain functions and macor to be defined to be defined on each platform ? Can we somehow add this to zephyr/include so that it would be obvious to anyone building zephyr that they need to add their own X,Y, Z for their target ?